test/unit/routes/first-time/test-about-you.js

```diff
@@ -1,78 +1,67 @@
 const proxyquire = require('proxyquire')
 const sinon = require('sinon')
 require('sinon-bluebird')
-const express = require('express')
-const bodyParser = require('body-parser')
-const mockViewEngine = require('../mock-view-engine')
+const ValidationError = require('../../../../app/services/errors/validation-error')
 
+var urlPathValidatorStub
 var stubInsertVisitor
-var stubAboutYouValidator
-var request
+var stubAboutYou
+var app
 
 describe('routes/first-time/about-you', function () {
-  var urlValidatorCalled = false
-
   beforeEach(function () {
+    urlPathValidatorStub = sinon.stub()
     stubInsertVisitor = sinon.stub()
-    stubAboutYouValidator = sinon.stub()
+    stubAboutYou = sinon.stub()
     var route = proxyquire('../../../../app/routes/first-time/about-you', {
       '../../services/data/insert-visitor': stubInsertVisitor,
-      '../../services/validators/first-time/about-you-validator': stubAboutYouValidator,
-      '../../services/validators/url-path-validator': function () { urlValidatorCalled = true }
+      '../../services/domain/about-you': stubAboutYou,
+      '../../services/validators/url-path-validator': urlPathValidatorStub
     })
 
-    var app = express()
-    app.use(bodyParser.urlencoded({ extended: false }))
-    mockViewEngine(app, '../../../app/views')
-    route(app)
-    request = supertest(app)
-    urlValidatorCalled = false
+    app = routeHelper.buildApp(route)
   })
 
   describe('GET /first-time/:dob/:relationship/:benefit/:reference', function () {
-    it('should respond with a 200 for valid path parameters', function (done) {
-      request
+    it('should respond with a 200 for valid path parameters', function () {
+      return supertest(app)
         .get('/first-time/1980-01-01/partner/income-support/1234567')
         .expect(200)
-        .end(function (error, response) {
-          expect(error).to.be.null
-          expect(urlValidatorCalled).to.be.true
-          done()
+        .expect(function () {
+          sinon.assert.calledOnce(urlPathValidatorStub)
         })
     })
   })
 
   describe('POST /first-time/:dob/:relationship/:benefit/:reference', function () {
-    it('should persist data and redirect to /application-submitted/:reference for valid data', function (done) {
+    it('should persist data and redirect to /application-submitted/:reference for valid data', function () {
       var reference = '1234567'
+      var aboutYou = {}
       stubInsertVisitor.resolves()
-      stubAboutYouValidator.returns(false)
+      stubAboutYou.returns(aboutYou)
 
-      request
+      return supertest(app)
         .post(`/first-time/1980-01-01/partner/income-support/${reference}`)
         .expect(302)
-        .end(function (error, response) {
-          expect(error).to.be.null
-          expect(urlValidatorCalled).to.be.true
-          expect(stubAboutYouValidator.calledOnce).to.be.true
+        .expect(function (response) {
+          sinon.assert.calledOnce(urlPathValidatorStub)
```

Comment:
There is already a unit test dedicated to testing for the urlPathValidator being called. I don't think we should check for it in each test.